python/cudf/cudf/core/series.py

```diff
@@ -3591,6 +3592,81 @@ def second(self):
         """
         return self._get_dt_field("second")
 
+    @property  # type: ignore
+    @_cudf_nvtx_annotate
+    def millisecond(self):
+        """
+        The millisecond of the datetime.
+
+        Examples
+        --------
+        >>> import pandas as pd
+        >>> import cudf
+        >>> datetime_series = cudf.Series(pd.date_range("2000-01-01",
+        ...         periods=3, freq="ms"))
+        >>> datetime_series
+        0    2000-01-01 00:00:00.000
+        1    2000-01-01 00:00:00.001
+        2    2000-01-01 00:00:00.002
+        dtype: datetime64[ns]
+        >>> datetime_series.dt.millisecond
+        0    0
+        1    1
+        2    2
+        dtype: int16
+        """
+        return self._get_dt_field("millisecond")
+
+    @property  # type: ignore
+    @_cudf_nvtx_annotate
+    def microsecond(self):
```

Comment:
The behavior of `microsecond` probably differs from pandas. Pandas doesn't have a `millisecond`, and it returns `milliseconds * 1000 + microseconds` for this function.

```python
>>> datetime_series = pd.Series(pd.date_range("2000-01-01", periods=1_000_000, freq="us")[-3:])
>>> datetime_series.dt.microsecond
0    999997
1    999998
2    999999
```

We should consider one (or more) of the following:
- align with pandas (remove the `millisecond` property and keep only `microsecond` as `milliseconds * 1000 + microseconds`)
- document this difference in behavior in both `millisecond` and `microsecond`
- return milliseconds for the `millisecond` property (property not in pandas) and `milliseconds * 1000 + microseconds` for the `microsecond` property (matches pandas)

Reply:
I think this is a bug on pandas' side, because `microsecond` is the only property that has this issue, I've opened an issue: https://github.com/pandas-dev/pandas/issues/49073.

Reply:
I've also xfailed `microsecond` for now, since we have test coverage on libcudf side I didn't add a hand-written validation for it, we can remove the xfail once the issue is resolved on pandas side. Let me know what you think.

Reply:
Had discussions offline and decided to mimic pandas' behavior.